pkg/adapterManager/manager.go

```diff
@@ -149,10 +159,27 @@ func (m *Manager) Quota(ctx context.Context, requestBag *attribute.MutableBag, r
 	return qmr, o
 }
 
+// Preprocess dispatches to the set of aspects that must run before any other
+// configured aspects.
+func (m *Manager) Preprocess(ctx context.Context, requestBag, responseBag *attribute.MutableBag) rpc.Status {
+	return m.dispatch(ctx, requestBag, responseBag, m.preprocessKindSet,
+		func(executor aspect.Executor, eval expr.Evaluator) rpc.Status {
+			ppw := executor.(aspect.PreprocessExecutor)
+			result, rpcStatus := ppw.Execute(requestBag, eval)
+			if status.IsOK(rpcStatus) {
+				if err := responseBag.Merge(result.Attrs); err != nil {
+					// TODO: better error messages that push internal details into debuginfo messages
+					rpcStatus = status.WithInternal("The results from the request preprocessing could not be merged.")
+				}
+			}
+			return rpcStatus
+		})
+}
+
 type invokeExecutorFunc func(executor aspect.Executor, evaluator expr.Evaluator) rpc.Status
 
 // dispatch resolves config and invokes the specific set of aspects necessary to service the current request
-func (m *Manager) dispatch(ctx context.Context, requestBag *attribute.MutableBag, responseBag *attribute.MutableBag,
+func (m *Manager) dispatch(ctx context.Context, requestBag, responseBag *attribute.MutableBag,
```

Comment:
Yeah, I thought about having Resolve short-circuit for kinds in the preprocess kind set and/or making a ResolveGlobal.

I'm concerned about having a manager by a different name and stuck in a strange part of the codebase. But I'm also concerned about trying to explain why some adapters don't have corresponding aspects in the config file (and why you can't have cross adapter config for them). I'm worried it will be a strange UX.

I'll noodle on this for awhile this weekend. Perhaps there is an approach that will be OK for alpha, but will need tweaking for beyond.

Let's discuss again Monday. SG?